versioning/BRM/extentmap.cpp

```diff
@@ -3253,97 +3071,87 @@ void ExtentMap::logAndSetEMIndexReadOnly(const std::string& funcName)
 //   startBlockOffset-starting block of the created extent
 // returns starting LBID of the created extent.
 //------------------------------------------------------------------------------
-LBID_t ExtentMap::_createColumnExtentExactFile(uint32_t size, int OID,
-        uint32_t  colWidth,
-        uint16_t  dbRoot,
-        uint32_t  partitionNum,
-        uint16_t  segmentNum,
-        execplan::CalpontSystemCatalog::ColDataType colDataType,
-        uint32_t& startBlockOffset)
+LBID_t ExtentMap::_createColumnExtentExactFile(uint32_t size, int OID, uint32_t colWidth, uint16_t dbRoot,
+                                               uint32_t partitionNum, uint16_t segmentNum,
+                                               execplan::CalpontSystemCatalog::ColDataType colDataType,
+                                               uint32_t& startBlockOffset)
 {
-    auto lastIndEmptyIndEmptyInd = _createExtentCommonSearch(OID, dbRoot,
-        partitionNum, segmentNum);
-    LastExtentIndexT lastExtentIndex = lastIndEmptyIndEmptyInd.first;
-    EmptyEMEntry emptyEMEntry = lastIndEmptyIndEmptyInd.second;
+    uint32_t highestOffset = 0;
+    LBID_t startLBID = getLBIDsFromFreeList(size);
+    EMEntry* lastEmEntry = nullptr;
 
-    if (emptyEMEntry == -1)
+    ASSERT(fExtentMapRBTree);
+
+    for (auto& emEntryPair : *fExtentMapRBTree)
```

Comment:
IMHO we should revert this chunk and apply Index search here b/c we increase the complexity from O(1) to O(N) in the worst case.